cmd/tcl/testworkflow-toolkit/artifacts/tar_stream.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2024 Testkube.
+//
+// Licensed as a Testkube Pro file under the Testkube Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//	https://github.com/kubeshop/testkube/blob/main/licenses/TCL.txt
+
+package artifacts
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/fs"
+	"sync"
+)
+
+type tarStream struct {
+	r  io.ReadCloser
```

Comment:
can we name it somehow - quite unreadable
e.g. tarCachedProcessor have usual names